### PR TITLE
Feature/mnist

### PR DESCRIPTION
**Pull request by felixmzd**

### CODE REVIEW

File: iclr_wrap_up/datasets/mnist.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    X_test = X_test * 2.0 - 1.0
+
+    Y_train = keras_utils.to_categorical(y_train, nb_classes).astype('float32')
+    Y_test = keras_utils.to_categorical(y_test, nb_classes).astype('float32')
+
+    Dataset = namedtuple('Dataset', ['X', 'Y', 'y', 'nb_classes'])
+    training = Dataset(X_train, Y_train, y_train, nb_classes)
+    test = Dataset(X_test, Y_test, y_test, nb_classes)
+
+    del X_train, X_test, Y_train, Y_test, y_train, y_test
```
Comment (JarnoRFB) [inline iclr_wrap_up/datasets/mnist.py]: No variables need to be deleted. When the function returns, all local variables are deleted anyway.

File: iclr_wrap_up/experiment.py [hunk 2]
```diff
@@ -113,7 +113,11 @@ def plot_infoplane(measures, architecture_name, infoplane_measure, epochs, activ
         ax.plot(xmvals, ymvals, color=color, alpha=0.1, zorder=1)
         ax.scatter(xmvals, ymvals, s=20, facecolors=color, edgecolor='none', zorder=2)
 
-    ax.set(xlim=[0, 12], ylim=[0, 1], xlabel='I(X;M)', ylabel='I(Y;M)')
+
+    if(dataset == "datasets.mnist" or dataset ==  "datasets.fashion_mnist"):
+        ax.set(xlim=[0, 14], ylim=[0, 3.5], xlabel='I(X;M)', ylabel='I(Y;M)')
```
Comment (JarnoRFB) [inline iclr_wrap_up/experiment.py]: If labels are same for both cases, dont put them in if block.

Comment (JarnoRFB) [inline iclr_wrap_up/experiment.py]: Could the limits be computed from the dataset?

File: iclr_wrap_up/datasets/mnist.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    X_train = np.reshape(X_train, [X_train.shape[0], -1]).astype('float32') / 255.
+    X_test = np.reshape(X_test, [X_test.shape[0], -1]).astype('float32') / 255.
+
+    X_train = X_train * 2.0 - 1.0
+    X_test = X_test * 2.0 - 1.0
+
+    Y_train = keras_utils.to_categorical(y_train, nb_classes).astype('float32')
+    Y_test = keras_utils.to_categorical(y_test, nb_classes).astype('float32')
+
+    Dataset = namedtuple('Dataset', ['X', 'Y', 'y', 'nb_classes'])
```
Comment (JarnoRFB) [inline iclr_wrap_up/datasets/mnist.py]: Maybe rather `n_classes` instead of `nb_classes` as this would be consistent with `n_runs`.

File: iclr_wrap_up/datasets/mnist.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    Returns:
+        Returns two namedtuples, the first one containing training
+        and the second one containing test data respectively. Both come with fields X, y and Y:
+        - X is the data
+        - y is class, with numbers from 0 to 9
+        - Y is class, but coded as a 10-dim vector with one entry set to 1 at the column index corresponding to the class
+    """
+    nb_classes = 10
+    (X_train, y_train), (X_test, y_test) = keras.datasets.mnist.load_data()
+    X_train = np.reshape(X_train, [X_train.shape[0], -1]).astype('float32') / 255.
```
Comment (JarnoRFB) [inline iclr_wrap_up/datasets/mnist.py]: Better with an explicit 0 behind the dot, ie `255.0`

File: iclr_wrap_up/utils.py [hunk 5]
```diff
@@ -1,102 +1,46 @@
...
+
+    Returns:
+        Both arrays shuffled in the same way
+    """
     assert len(a) == len(b)
     p = np.random.permutation(len(a))
     return a[p], b[p]
 
+
 def data_shuffle(data_sets_org, percent_of_train, min_test_data=80, shuffle_data=False):
```
Comment (JarnoRFB) [inline iclr_wrap_up/utils.py]: This code is a little opaque to me. Maybe this could be replaced with a preimplemented method from sklearn like http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html
